### PR TITLE
Add description for BAT board on documentation

### PR DESCRIPTION
**Pull request by MSECode**
Added documentation regarding the BAT board and explaining the meaning of the status sent to the Display by the board.
Reorganize then the section regarding the type of batteries of the robot.

### CODE REVIEW

File: docs/battery_boards/index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Battery Documentation
+Here you can find information about different boards related to the battery in the various versions of the robot and some related resources.
+
+[**BCB Board Documentation**](./bcb_board/bcb_bluetooth_protocol.md)
+
+[**BAT Board Documentation**](./bat_board/bat_protocol.md)
```
Comment (pattacini) [inline docs/battery_boards/index.md:6]:
```suggestion
- [**BCB Board Documentation**](./bcb_board/bcb_bluetooth_protocol.md)
- [**BAT Board Documentation**](./bat_board/bat_protocol.md)
```

File: docs/battery_boards/bat_board/bat_protocol.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# BAT board Protocol
+
+## Introduction
+This page describes the `BAT` board and its protocol. Basically, this board is responsible for the management of the data and signals coming from the battery pack in `R1`, `iCub` and `ergoCub` robots. Moreover, it should be noted that in the latter robots mentioned, it is always coupled with the `BMS` baord, which is responsible for a more fine tuning and managing of the battery pack status.
```
Comment (pattacini) [inline docs/battery_boards/bat_board/bat_protocol.md:4]:
```suggestion
This page describes the `BAT` board and its protocol. Basically, this board is responsible for the management of the data and signals provided by the battery pack in `R1`, `iCub`, and `ergoCub` robots. Moreover, it should be noted that in these robots it is always coupled with the `BMS` baord, which is responsible for fine-tuning and managing the battery pack status.
```

File: docs/battery_boards/bat_board/bat_protocol.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+In general, as shown in the figures below (which actually show the connection between the BAT and EMS board in the base of `R1/SN003`), the `BAT` board can be connected to an EMS board through the specific `CAN` connector, which is responsible for reading the CAN frames coming from the `BAT`.
```
Comment (pattacini) [inline docs/battery_boards/bat_board/bat_protocol.md:5]:
```suggestion
In general, as shown in the figures below (illustrating the connection between the BAT and EMS boards in the base of `R1/SN003`), the `BAT` can be connected to an EMS board through the `CAN` connector devoted to receiving the CAN frames sent out by the `BAT`.
```

File: docs/battery_boards/bat_board/bat_protocol.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+<td><img width="500" height="500" src="../assets/bat-2-ems-R1-connection-01.png"/></td>
+<td><img width="500" height="500" src="../assets/bat-2-ems-R1-connection-02.png"/></td>
+</tr></table>
+</center>
+
+Specifically, at the current status of the developments (July 2023), the `BAT` board can send the following data:
+
+- Battery Pack Voltage in Volt
+- Battery Pack Charge in $\%$
+- Battery Pack Status (details about it will follow)
```
Comment (pattacini) [inline docs/battery_boards/bat_board/bat_protocol.md:18]:
```suggestion
Specifically, at the current status of the development (July 2023), the `BAT` board can send the following data:

- Battery Pack Voltage in Volt.
- Battery Pack Charge in $\%$.
- Battery Pack Status (detail will follow).
```

File: docs/battery_boards/bat_board/bat_protocol.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+
+### Communication characteristics
+
+The pieces of information detailed above are sent by the `BAT` board with a FIFO cycle of 1ms. Then, the `EMS` board handles the CAN frames sent by the `BAT`, parses them and finally forwards them to the higher level of the `yarprobotinterface` at the specific port defined in the configuration files with a frequence o 1Hz.
```
Comment (pattacini) [inline docs/battery_boards/bat_board/bat_protocol.md:23]:
```suggestion
The pieces of information detailed above are sent by the `BAT` board with a FIFO cycle of 1 ms. Then, the `EMS` board handles the CAN frames sent by the `BAT`, parses them, and finally forwards them to the higher level of the `yarprobotinterface` at the specific port defined in the configuration files with a frequency of 1 Hz.
```

File: docs/battery_boards/bat_board/bat_protocol.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+Furthermore, those info are also sent directly from the `BAT` to the display attached to the robot, every 10ms.
```
Comment (pattacini) [inline docs/battery_boards/bat_board/bat_protocol.md:24]:
```suggestion
Furthermore, that info is also sent directly from the `BAT` to the display attached to the robot, every 10 ms.
```

Remember that info is uncountable.

File: docs/battery_boards/bat_board/bat_protocol.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    `(DCDC_status_B << 8 ) | DCDC_status_A`
+
+
+### Data displayed on the yarp port
+
+It should be noted that as things now the user will see the data at the specific yarp port defined in the configuration file as follows:
+
+- Voltage will be displayed in the Volts
+- Battery charge will be displayed in $\%$
+- Status will be displayed as an integer of 16 bits (where we fill only the first 10), which can be interpreted following the tables above
```
Comment (pattacini) [inline docs/battery_boards/bat_board/bat_protocol.md:85]:
```suggestion
The user gets the data from a specific YARP port defined in the configuration file as follows:

- Voltage is displayed in the Volts.
- Battery charge is displayed in $\%$.
- Status is displayed as a 16-bit integer (only the first 10 bits are valid), whose mapping adheres to the tables above.
```

File: docs/battery_boards/bat_board/bat_protocol.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+Moreover, when the boards start a **DEBUG** message with the initial values of the status (converted to the description strings) is send to the `yarprobotinterface`.
```
Comment (pattacini) [inline docs/battery_boards/bat_board/bat_protocol.md:87]:
```suggestion
Moreover, at start-up, a **DEBUG** message with the initial values of the status (converted to the description strings) is sent to `yarprobotinterface`. 
```

File: docs/battery_boards/bat_board/bat_protocol.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+### Data displayed on the yarp port
+
+It should be noted that as things now the user will see the data at the specific yarp port defined in the configuration file as follows:
+
+- Voltage will be displayed in the Volts
+- Battery charge will be displayed in $\%$
+- Status will be displayed as an integer of 16 bits (where we fill only the first 10), which can be interpreted following the tables above
+
+Moreover, when the boards start a **DEBUG** message with the initial values of the status (converted to the description strings) is send to the `yarprobotinterface`. 
+Then, each time the status changes a **DEBUG** message is send to the `yarprobotinterface` that shows only, as description strings, the value of the bits changed with respect to the previous status.
```
Comment (pattacini) [inline docs/battery_boards/bat_board/bat_protocol.md:88]:
```suggestion
Then, each time the status changes, a **DEBUG** message is sent to `yarprobotinterface`, which in turn prints out a description only about the values of those bits that have switched.
```

**Review by pattacini — CHANGES_REQUESTED** [review]
Hi @MSECode 

Thanks for your contribution.
There are several points to address, which I deem make it read more fluently.

For each, I've posted suggestions that you may review and accept.

**Comment by MSECode** [comment]
Thanks @pattacini for the review and suggestions

File: docs/battery_boards/bat_board/bat_protocol.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    Possible status | Start-up phase(1)/stable operation(0) | Start-up phase/stable operation |
+
+- Final status shown at the port is equal to: 
+
+    `(DCDC_status_B << 8 ) | DCDC_status_A`
+
+
+### Data displayed on the yarp port
+
+The user gets the data from a specific YARP port defined in the configuration file as follows:
```
Comment (pattacini) [inline docs/battery_boards/bat_board/bat_protocol.md:81]:
```suggestion
The user gets the data from a specific YARP port defined in the configuration file. Here's the format:
```

**Review by pattacini — APPROVED** [review]

**Pull request merged**